### PR TITLE
Implement WordpressFeed component

### LINKED ISSUE #268: Create a WordpressFeed component
Requires #267

Create a component that will display a wordpress rss feed. The component should accept a title. Optionally a footer with a text and a link to the site.

dashboard_widget_redesign_ __7549_ _yoast_seo___waffle_io

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:

* Added WordpressFeed component

## Test instructions

This PR can be tested by following these steps:

* Replace the contents of `ContentAnalysis` with the following content:
```js
import React from "react";
import styled from "styled-components";

import WordpressFeed from "../../DashboardWidget/components/WordpressFeed";

export const ContentAnalysisContainer = styled.div`
   min-height: 700px;
   background-color: white;
`;

export const WordpressFeedContainer = styled.div`
	border: 1px solid black;
	width: 400px;
`;

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
export default function ContentAnalysis() {
	const feed = {
		link: "https://www.yoast.com",
		title: "Feed title",
		items: [
			{
				title: "Wordpress SEO",
				link: "https://www.yoast.com/1",
				description: "Some arbitrary description any blog post could have",
			},
			{
				title: "Wordpress SEO",
				link: "https://www.yoast.com/2",
				description: "Some arbitrary description any blog post could have",
			},
		],
	};
	return (
		<ContentAnalysisContainer>
			<WordpressFeedContainer>
				<WordpressFeed
					feed={ feed }
					title={ "Yoast SEO" }
					footerHtml="Go to Yoast's SEO blog now!" />
			</WordpressFeedContainer>
		</ContentAnalysisContainer>
	);
}
```
Fixes #268

### CODE REVIEW

**Comment by herregroen** [comment]
CR Done 👍 .

@afercia Could you acceptance test this and double check the accessibility?

File: composites/Plugin/DashboardWidget/components/WordpressFeed.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import colors from "../../../../style-guide/colors.json";
+
+/**
+ * @typedef  {Object}     Feed
+ * @property {string}     title       The title of the website.
+ * @property {string}     description A description of the website.
+ * @property {string}     link        A link to the website.
+ * @property {FeedItem[]} items     The items in the feed.
```
Comment (afercia) [inline composites/Plugin/DashboardWidget/components/WordpressFeed.js]: alignment of the description

File: composites/Plugin/DashboardWidget/components/WordpressFeed.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+const WordpressFeedList = styled.ul`
+	margin: 0;
+	list-style: none;
+	padding: 0;
+`;
+
+const WordpressFeedLink = styled.a`
+`;
+
```
Comment (afercia) [inline composites/Plugin/DashboardWidget/components/WordpressFeed.js]: not sure why we should use a styled component that doesn't set any style; why not use directly an `a` element in the JSX?

File: composites/Plugin/DashboardWidget/components/WordpressFeed.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+						key={ item.link }
+						title={ item.title }
+						link={ item.link }
+						description={ item.description } />
+				) ) }
+			</WordpressFeedList>
+			{ props.footerHtml &&
+				<WordpressFeedFooter>
+					<WordpressFeedLink
+						href={props.feedLink ? props.feedLink : props.feed.link}
```
Comment (afercia) [inline composites/Plugin/DashboardWidget/components/WordpressFeed.js]: missing spaces within brackets

**Comment by afercia** [comment]
Acceptance 🚧 

- a few minor coding standard issues
- I'm not sure (because I really don't know) about the two bit docblocks at the beginning, I guess it's OK?
- based on the design, the link in the footer should have an icon; I guess this should be addressed during implementation?

a11y:
- all links that use `target="_blank"` should inform users the link is going to open in a new tab/window. In My Yoast we have a specific component for this, that's meant to add some visually hidden text _inside_ the link 
- let's add `role="list"` to the list (redundant but it's a workaround for a VoiceOver bug we're using in other places too)

**Comment by Xyfi** [comment]
@afercia 
* The doc blocks are meant to specify what shape the component expects the component to be.
* The icon will be addressed during implementation, that is why the footer accepts HTML.

I will address the `a11y` and coding standards issues.

**Comment by afercia** [comment]
Thanks for the clarification!

**Comment by afercia** [comment]
On My Yoast: `client/src/components/NewTabMessage.js`

**Comment by afercia** [comment]
Just realized that all links that open a new window should also use ` rel="noopener noreferrer"`

**Comment by afercia** [comment]
CR and acceptance 👍 

Please make sure a screen-reader-text with a `(Opens in a new browser tab)` gets then passed for the footer link in the `footerHtml` prop.

**Pull request merged**